### DISCUSSION TITLE
Error: P1001: Can't reach database server at `MY-CONNECTION-STRING` (vercel | neon | supabase | railway)

### DISCUSSION THREAD

**Discussion by Naeem-gg** [prisma/orm#23589] [Q&A]
### Bug description
![image](https://github.com/prisma/prisma/assets/91608646/5cd93593-38a9-4aee-906b-77cb312d2c65)

Hi there! I think there's a problem with prisma with serverless postgresql, I tried this on vercel postgres, supabase, railway and Neon and found this issue with all of mentioned services where it shows the same error as you can see in uploaded image. It happens very less with vercel postgres compared to others. How it gets resolved? well you need to sign in to the service dashboard and do something with databse either view some records or edit it doesn't matter you just need to interact with it and then come back to terminal and run any prisma command it will work fine. I never encountered the same with drizzle. So that's clearly something wrong with prisma and cold start of serverless postgres

### How to reproduce

There's no perticular way to reproduce it but It happens with Neon the most. So create a db with Neon and continue working on it (while all the browsers where neon is logged in are closed to make sure its not running in bg) and you'll get the exact same error when you fire prisma commands `db push/pull` or even `migrate dev`

### Prisma information

```prisma
// Add your schema.prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model User {
  id        String     @id @default(cuid())
  email     String     @unique
  name      String
  password  String
  unicode   String
  createdAt DateTime   @default(now())
  updatedAt DateTime   @updatedAt
}
```
```ts
// Add your code using Prisma Client
import { PrismaClient } from "@prisma/client";

declare global {
  namespace NodeJS {
    interface Global {}
  }
}
interface CustomNodeJsGlobal extends NodeJS.Global {
  prisma: PrismaClient;
}
declare const global: CustomNodeJsGlobal;

const prisma = global.prisma || new PrismaClient();

if (process.env.NODE_ENV === "development") global.prisma = prisma;

export default prisma;

```

### Environment & setup

prisma                  : 5.11.0
@prisma/client          : 5.11.0
Computed binaryTarget   : windows
Operating System        : win32
Architecture            : x64
Node.js                 : v20.9.0
Studio                  : 0.499.0

### Prisma Version

```
Environment variables loaded from .env
prisma                  : 5.11.0
@prisma/client          : 5.11.0
Computed binaryTarget   : windows
Operating System        : win32
Architecture            : x64
Node.js                 : v20.9.0
Studio                  : 0.499.0
```

**Comment by Naeem-gg** [comment]
I'm using Nextjs v14.1.4 + vercel postgres with mentioned versions of prisma and client (if it helps)

**Reply by kameyaw14** [reply]
For me I just changed my wifi connection and it worked

**Reply by JayfaAnjum** [reply]
yeah it worked for me also

**Comment by ludralph** [comment]
Hi @Naeem-gg 👋 

The error you're encountering, `P1001: Can't reach database server`, is a common issue when using Prisma with serverless databases like Neon. This error typically occurs when the connection created by Prisma Client times out before the serverless compute is activated.

Serverless databases like Neon have two main states: Active and Idle. If there is no query activity for a certain period (like 5 minutes for Neon), the database goes into an idle state. When you try to connect to an idle compute from Prisma ORM, Neon automatically activates it. However, this activation can take a few seconds and added latency can result in a connection timeout.

To address this issue, you can adjust your connection string by adding a `connect_timeout` parameter. This parameter defines the maximum number of seconds to wait for a new connection to be opened. The default value is 5 seconds. A higher setting should provide the time required to avoid connection timeout issues. For example:

```
DATABASE_URL=postgres://daniel:<password>@ep-mute-rain-952417.us-east-2.aws.neon.tech/neondb?connect_timeout=10
```
A `connect_timeout` setting of 0 means no timeout.

Another possible cause of connection timeouts is Prisma ORM's connection pool, which has a default timeout of 10 seconds. If you are still experiencing connection timeouts, you can try increasing this limit (in addition to the `connect_timeout` setting described above) by setting the `pool_timeout` parameter to a higher value.

See this section of the [documentation](https://www.prisma.io/docs/orm/overview/databases/neon#resolving-connection-timeouts) on resolving connection timeout with Neon.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by Naeem-gg** [reply]
> Hi @Naeem-gg 👋 
> 
> The error you're encountering, `P1001: Can't reach database server`, is a common issue when using Prisma with serverless databases like Neon. This error typically occurs when the connection created by Prisma Client times out before the serverless compute is activated.
> 
> Serverless databases like Neon have two main states: Active and Idle. If there is no query activity for a certain period (like 5 minutes for Neon), the database goes into an idle state. When you try to connect to an idle compute from Prisma ORM, Neon automatically activates it. However, this activation can take a few seconds and added latency can result in a connection timeout.
> 
> To address this issue, you can adjust your connection string by adding a `connect_timeout` parameter. This parameter defines the maximum number of seconds to wait for a new connection to be opened. The default value is 5 seconds. A higher setting should provide the time required to avoid connection timeout issues. For example:
> 
> ```
> DATABASE_URL=postgres://daniel:<password>@ep-mute-rain-952417.us-east-2.aws.neon.tech/neondb?connect_timeout=10
> ```
> A `connect_timeout` setting of 0 means no timeout.
> 
> Another possible cause of connection timeouts is Prisma ORM's connection pool, which has a default timeout of 10 seconds. If you are still experiencing connection timeouts, you can try increasing this limit (in addition to the `connect_timeout` setting described above) by setting the `pool_timeout` parameter to a higher value.
> 
> See this section of the [documentation](https://www.prisma.io/docs/orm/overview/databases/neon#resolving-connection-timeouts) on resolving connection timeout with Neon.
> 
> If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
> 
> Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

Hey @ludralph 👋 thanks for your response, please let me check these parameters that you suggested. I just have one small question here, is it safe to override the default timeouts?

**Reply by Naeem-gg** [reply]
Hey @ludralph its still the same with connect_timeout=20 as well

**Reply by ludralph** [reply]
Can you try increasing both `connect_timeout` and `pool_timeout`. According to the [documentation on Neon](https://www.prisma.io/docs/orm/overview/databases/neon#resolving-connection-timeouts), you can try adjusting both the `connect_timeout` and `pool_timeout` parameters:
```
DATABASE_URL=postgres://daniel:<password>@ep-mute-rain-952417.us-east-2.aws.neon.tech/neondb?connect_timeout=15&pool_timeout=15
```

**Reply by mdkaifansari04** [reply]
Thanks @ludralph, it worked for me, I just increase the `connect_timeout=20` and it worked. Guys we can defenitly modify the connection_timeout as it was mentioned in the [prisma-neon docs](https://www.prisma.io/docs/orm/overview/databases/neon#resolving-connection-timeouts), so this is safe.

**Reply by cerastvs** [reply]
@ludralph thanks bro it worked for me!

**Comment by 98hub** [comment]
I am facing the same issue rn.
Is there anyone who has solved it?

**Reply by Naeem-gg** [reply]
Tried multiple methods but still faced the same issue, at last I switched to drizzle.

**Reply by TemryL** [reply]
Got the same issue. Turns out it was due to my wifi connection. I was connected on eduroam.

**Reply by Divyanshu1020** [reply]
> Got the same issue. Turns out it was due to my wifi connection. I was connected on eduroam.

facing same issue with jio air fiber

**Reply by YuZhang-steven** [reply]
> Got the same issue. Turns out it was due to my wifi connection. I was connected on eduroam.

same issue, using the school wifi. I am wondering if there is some setting to bypass it

**Reply by Lukas-Blom** [reply]
Hi I also switched to my mobile internet wifi and it worked. Other wifi might have firewalls that blocks this type of connection. Hope this helps.

**Reply by Blake-Larson** [reply]
I was on my work's VPN, disconnected and it worked.

**Reply by jv18creator** [reply]
@Divyanshu1020 how did you solve it?

**Reply by Divyanshu1020** [reply]
Just change dns setting, use Google dns server
In my opinion 
By default computer don't have any dns setting so by default network use router or modem dns setting that's why it's is not working with wifi network

**Reply by salman-node** [reply]
same issue RESOLVED just my adding  ?connect_timeout=15&pool_timeout=15  in url

**Comment by Reinhartsamuel** [comment]
I tried to connect prisma and neon db for ever but never succeeded, so I moved to some other managed postgresql service:
- koyeb : same problem
- aiven : success after 3 attempts

So I just use aiven instead of neon

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;

**Comment by abdullahilashik** [comment]
Removing the **sslmode=require** resolved the issue for me.

**Reply by Akhil7K** [reply]
I don't believe it is a right thing to do.

**Comment by theogenejr** [comment]
The issue you are facing has to deal with the formatting of your DATABASE_URL value in your .env file. What you need to do is add a ? and a timeout command after your url

update the value in your .env file like this

`DATABASE_URL = 'url + ?sslaccept=strict&connect_timeout=500'`

Forexample

`..........us-east-2.aws.neon.tech/neondb?sslmode=require?sslaccept=strict&connect_timeout=500`

**Reply by bryanpascuadizon** [reply]
This worked for me. thanks!

**Comment by Tiru-99** [comment]
I’ve run into this issue multiple times, and here’s what usually helps:

✅ 1️⃣ Increase the connection timeout — sometimes the default is too low, and bumping it up solves the problem. You can do it by adding ` sslaccept=strict&connect_timeout=500` at the end of your connection string

✅ 2️⃣ (This works ~70% of the time!) — it’s a network issue.
Try switching your internet connection or disabling your VPN — trust me, it works like magic! ✨

**Reply by davistran86** [reply]
thanks, connect_timeout solved my current problem 👍

**Comment by kadhirvelm** [comment]
On the off chance this helps someone else, I was running into the same issue for weeks on an EC2. I tried increasing the connect_timeout and it didn't work, but interestingly I could get it to work from my local machine. So I tried an nslookup on my EC2 and on my local machine to see if it was resolving the addresses differently. Which as it turns out, my EC2 had some suspicious looking IPv6 addresses, which my local machine did not.

So I temporarily disabled the IPv6 stuff on my EC2 via 
```
sudo sysctl -w net.ipv6.conf.all.disable_ipv6=1
sudo sysctl -w net.ipv6.conf.default.disable_ipv6=1
```

And tried again with `connect_timeout=30` and it worked great after that

**Comment by yihui-zhou** [comment]
not working for me adding timeout

**Comment by nestcub** [comment]
increasing connection timeout doesnt work for me.
resetting password didnt work for me.
disabling 5432 (ipv6) on the remote server didnt work for me.


what else i can do?

**Comment by Shubhofficial1** [comment]
Sometimes the wifi network is somehoe blocking the request. Try switching to some other network just to double check if it is realted to firewall issue. Otherwise use the connection_timeout solution

**Comment by Bala-Prakash-M** [comment]
For me it worked after VPN

**Reply by ichavesbr** [reply]
Same here. I use Proton VPN, and it always gives me issues with Prisma + Neon. I closed the VPN, and it worked.
